plugin/src/cli_utils/localpvlvm/node/types.rs

```diff
@@ -0,0 +1,179 @@
+use k8s_openapi::NamespaceResourceScope;
+use kube::{api::ObjectMeta, Resource, ResourceExt};
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
+
+impl Resource for LvmNode {
+    type DynamicType = ();
+
+    fn kind(_: &Self::DynamicType) -> Cow<'_, str> {
+        Cow::Borrowed("LVMNode")
+    }
+
+    fn group(_: &Self::DynamicType) -> Cow<'_, str> {
+        Cow::Borrowed("local.openebs.io")
+    }
+
+    fn version(_: &Self::DynamicType) -> Cow<'_, str> {
+        Cow::Borrowed("v1alpha1")
+    }
+
+    fn plural(_: &Self::DynamicType) -> Cow<'_, str> {
+        Cow::Borrowed("lvmnodes")
+    }
+
+    fn meta(&self) -> &ObjectMeta {
+        &self.metadata
+    }
+
+    fn meta_mut(&mut self) -> &mut ObjectMeta {
+        &mut self.metadata
+    }
+
+    type Scope = NamespaceResourceScope;
+}
+
+/// LVMNode struct mirroring the crd.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub(crate) struct LvmNode {
+    #[serde(alias = "apiVersion")]
+    api_version: String,
+    kind: String,
+    metadata: ObjectMeta,
+    #[serde(alias = "volumeGroups")]
+    volume_groups: Vec<VolumeGroup>,
+}
+
+/// Lvm Volume group.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub(crate) struct VolumeGroup {
+    #[serde(alias = "allocationPolicy")]
+    allocation_policy: i32,
+    free: String,
+    #[serde(alias = "lvCount")]
+    lv_count: i32,
+    #[serde(alias = "maxLv")]
+    max_lv: i32,
+    #[serde(alias = "maxPv")]
+    max_pv: i32,
+    #[serde(alias = "metadataCount")]
+    metadata_count: i32,
+    #[serde(alias = "metadataFree")]
+    metadata_free: String,
+    #[serde(alias = "metadataSize")]
+    metadata_size: String,
+    #[serde(alias = "metadataUsedCount")]
```

Comment:
Why do we have alias for `VolumeGroup` but not `LvmVolumeGroup`?
What's the difference between them?

Reply:
VolumeGroup maps to `lvmnode.volume_groups = vec<VolumeGroup>`. This is part of the struct which maps to crd. In localpv-lvm , camel case to represent fields. Hence using alias to map it. 

`VolumeGroupRecord.vgs = Vec<LvmVolumeGroup>` implements the formatting trait.